spec/controllers/weather_controller_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe WeatherController, type: :controller do
+  describe 'temperature' do
+    before do
+      allow_any_instance_of(WeatherApi).to receive(:temp).and_return({
```

Comment:
Do you want to allow this? maybe you should test with the api directly and avoid hardcoding a temperature